versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecord.java

```diff
@@ -0,0 +1,40 @@
+package org.codehaus.mojo.versions.api.recording;
+
+import org.codehaus.mojo.versions.api.change.VersionChange;
+
+public interface ChangeRecord
+{
+    enum ChangeKind
+    {
+        COMPARE_DEPENDENCIES( "compareDependencies" ),
+        FORCE_RELEASES( "forceReleases" ),
+        UNLOCK_PARENT_VERSION( "unlockParentVersion" ),
+        UNLOCK_SNAPSHOT( "unlockSnapshot" ),
+        UPDATE_PARENT( "updateParent" ),
+        UPDATE_PROPERTY( "updateProperty" ),
+        USE_DEPENDENCY_VERSION( "useDependencyVersion" ),
+        USE_LATEST_RELEASES( "useLatestReleases" ),
+        USE_LATEST_SNAPSHOTS( "useLatestSnapshots" ),
+        USE_LATEST_VERSIONS( "useLatestVersions" ),
+        USE_NEXT_RELEASES( "useNextReleases" ),
+        USE_NEXT_SNAPSHOTS( "useNextSnapshots" ),
+        USE_NEXT_VERSIONS( "useNextVersions" ),
+        USE_RELEASES( "useReleases" );
```

Comment:
There is a list of label used as `kind` - for me it is more source of change ... maybe not needed at all

Reply:
What if we used this like the logger name in SLF4J and similar logger interfaces? Maybe make it a free-form text and use the class name with it, just like for loggers? What do you think?

Reply:
So if we use class name it will be rather then source of change than kind ...

Reply:
Maybe `kind` should be where version is changed: dependency, dependencyMangmen, plugin, pluginManagmen, properties, parent

Reply:
Good idea. +1

Reply:
done